wallet/wallet-rpc-lib/src/rpc/server_impl.rs

```diff
@@ -396,12 +398,19 @@ impl<N: NodeInterface + Clone + Send + Sync + Debug + 'static> WalletRpcServer f
     async fn get_balance(
         &self,
         account_arg: AccountArg,
+        utxo_states: Vec<RpcUtxoState>,
         with_locked: Option<WithLocked>,
     ) -> rpc::RpcResult<Balances> {
+        let utxo_states = if utxo_states.is_empty() {
+            UtxoStates::ALL
+        } else {
+            utxo_states.iter().map(UtxoState::from).fold(UtxoStates::NONE, |x, y| x | y)
```

Comment:
It's a 3rd place where this `fold` occurs. I'd add something like `UtxoStates::from_states` accepting `impl IntoIterator<Item = UtxoState>`.

Not insisting on this.